src/CodeGen_LLVM.cpp

```diff
@@ -2582,6 +2626,19 @@ void CodeGen_LLVM::visit(const Call *op) {
         Type dst = op->type;
         Type src = op->args[0].type();
         llvm::Type *llvm_dst = llvm_type_of(dst);
+        if (const Load *l = op->args[0].as<Load>()) {
+            if (Expr index = should_load_scalar(l, dst); index.defined()) {
+                Value *ptr =
+                    codegen_buffer_pointer(l->name, src.element_of(), index);
+                // Pessimistically assume only scalar element alignment.
+                LoadInst *load = builder->CreateAlignedLoad(
+                    llvm_dst, ptr, llvm::Align(l->type.bytes()));
+                // FIXME: can we emit better TBAA for constant indexes here?
+                add_tbaa_metadata(load, l->name, /*index=*/Expr());
```

Comment:
I think it's correct to use the original load index here, because the tbaa metadata is all in terms of the allocated type before the reinterpret.

Reply:
Ah no, wait, of course it is, because we really don't change which bytes we load.